components/mosart/src/riverroute/RtmMod.F90

```diff
@@ -1679,38 +2230,41 @@ subroutine Rtmrun(rstwr,nlend,rdate)
     ! --- convert TRunoff fields from m3/s to m/s before calling Euler
     !-----------------------------------
 
-!    if (budget_check) then
-       call t_startf('mosartr_budget')
-       do nt = 1,nt_rtm
-       do nr = rtmCTL%begr,rtmCTL%endr
-          budget_terms(20,nt) = budget_terms(20,nt) + TRunoff%qsur(nr,nt) + &
-             TRunoff%qsub(nr,nt) + TRunoff%qgwl(nr,nt)
-          budget_terms(29,nt) = budget_terms(29,nt) + TRunoff%qgwl(nr,nt)
-       enddo
-       enddo
-       call t_stopf('mosartr_budget')
-!    endif
-
     do nt = 1,nt_rtm
     do nr = rtmCTL%begr,rtmCTL%endr
        TRunoff%qsur(nr,nt) = TRunoff%qsur(nr,nt) / rtmCTL%area(nr)
        TRunoff%qsub(nr,nt) = TRunoff%qsub(nr,nt) / rtmCTL%area(nr)
        TRunoff%qgwl(nr,nt) = TRunoff%qgwl(nr,nt) / rtmCTL%area(nr)
+       TRunoff%qdem(nr,nt) = TRunoff%qdem(nr,nt) / rtmCTL%area(nr) !m3 to m added by Yuna 1/29/2018
     enddo
     enddo
 
     do ns = 1,nsub
 
-       if (wrmflag) then
-#ifdef INCLUDE_WRM
-          call t_startf('mosartr_euler')
-          call Euler_WRM()
-          call t_stopf('mosartr_euler')
-#else
-          call shr_sys_abort(trim(subname)//' ERROR WRM not compiled')
-#endif
+       ! this advances the model time one coupling period when ns=1
+       ! we want the advance in the subcycling because of the new_month flag
+       ! this should really be advancing delt per ns, not coupling_period per call, but
+       ! need to modify clock initialization and restart for that to be supported
+       ! and make sure a rational timestep doesn't lead to accumulating errors
+       call advance_timestep(ns)
+
+       call get_curr_date(yr, mon, day, tod)
+       ymd = yr*10000 + mon*100 + day
+       if (tod == 0 .and. masterproc) then
+          write(iulog,*) ' '
+          write(iulog,'(2a,i4,a,i10,i6)') trim(subname),' subcycling=',ns,': model date=',ymd,tod
+       endif
+     
+       if (inundflag) then
+          call t_startf('mosartr_inund_sim')
+!#ifdef INCLUDE_INUND
+          call MOSARTinund_simulate ( )
+!#endif     
+          call t_stopf('mosartr_inund_sim')
        else
           call t_startf('mosartr_euler')
+          ! debug 
+          write(iulog,*) 'clm-mosart subT: (call Euler) ns=', ns
```

Comment:
Please comment out or surround this line with `#ifndef -DNDEBUG ... #endif` to activate it for debug runs. A 5-day run produced 122721 writes to e3sm.log.

Reply:
@amametjanov open an issue about this.  Assign it to @hydrotian